lib/Ast.ml

```diff
@@ -1,223 +0,0 @@
-  | GetTarget
-  | ArrayExpr of 'e list
-  | RowVectorExpr of 'e list
-  | Paren of 'e
-  | Indexed of 'e * 'e index list
-
-and expression_untyped_metadata =
-  {expr_untyped_meta_loc: location sexp_opaque [@compare.ignore]}
-
-and expression_typed_metadata =
-  { expr_typed_meta_origin_type: originblock * unsizedtype
-  ; expr_typed_meta_loc: location sexp_opaque [@compare.ignore] }
-
-and untyped_expression =
-  | UntypedExpr of (untyped_expression expression * expression_untyped_metadata)
-
-and typed_expression =
-  | TypedExpr of (typed_expression expression * expression_typed_metadata)
-
-(* == Statements == *)
-and assignmentoperator =
-  | Assign
-  | PlusAssign
-  | MinusAssign
-  | TimesAssign
-  | DivideAssign
-  | EltTimesAssign
-  | EltDivideAssign
-  | ArrowAssign
-
-(* deprecated *)
-and 'e truncation =
-  | NoTruncate
-  | TruncateUpFrom of 'e
-  | TruncateDownFrom of 'e
-  | TruncateBetween of 'e * 'e
-
-and 'e printable = PString of string | PExpr of 'e
-
-and 'e sizedtype =
-  | SInt
-  | SReal
-  | SVector of 'e
-  | SRowVector of 'e
-  | SMatrix of 'e * 'e
-  | SArray of 'e sizedtype * 'e
-
-and 'e transformation =
-  | Identity
-  | Lower of 'e
-  | Upper of 'e
-  | LowerUpper of 'e * 'e
-  | LocationScale of 'e * 'e
-  | Ordered
-  | PositiveOrdered
-  | Simplex
-  | UnitVector
-  | CholeskyCorr
-  | CholeskyCov
-  | Correlation
-  | Covariance
-
-and ('e, 's) statement =
-  | Assignment of
-      { assign_identifier: identifier
-      ; assign_indices: 'e index list
-      ; assign_op: assignmentoperator
-      ; assign_rhs: 'e }
-  | NRFunApp of identifier * 'e list
-  | TargetPE of 'e
-  | IncrementLogProb of 'e
-  (* deprecated *)
-  | Tilde of
-      { arg: 'e
-      ; distribution: identifier
-      ; args: 'e list
-      ; truncation: 'e truncation }
-  | Break
-  | Continue
-  | Return of 'e
-  | ReturnVoid
-  | Print of 'e printable list
-  | Reject of 'e printable list
-  | Skip
-  | IfThenElse of 'e * 's * 's
-  | IfThen of 'e * 's
-  | While of 'e * 's
-  | For of
-      { loop_variable: identifier
-      ; lower_bound: 'e
-      ; upper_bound: 'e
-      ; loop_body: 's }
-  | ForEach of identifier * 'e * 's
-  (* TODO: maybe make separate nodes for matrix and array foreach loops? *)
-  | Block of 's list
-  | VDecl of 'e sizedtype * identifier
-  | VDeclAss of {sizedtype: 'e sizedtype; identifier: identifier; value: 'e}
-  | TVDecl of 'e sizedtype * 'e transformation * identifier
-  | TVDeclAss of
-      { tsizedtype: 'e sizedtype
-      ; transformation: 'e transformation
-      ; tidentifier: identifier
-      ; tvalue: 'e }
-  | FunDef of
-      { returntype: returntype
-      ; funname: identifier
-      ; arguments: (originblock * unsizedtype * identifier) list
-      ; body: 's }
-
-and statement_untyped_metadata =
-  {stmt_untyped_meta_loc: location sexp_opaque [@compare.ignore]}
-
-and statement_returntype =
-  | NoReturnType
-  | Incomplete of returntype
-  | Complete of returntype
```

Comment:
Need comments or renaming to indicate what these mean

Reply:
Done. 👍